env/env_encryption.cc

```diff
@@ -1215,19 +1215,24 @@ Status CTREncryptionProvider::CreateCipherStream(
   Slice iv;
   decodeCTRParameters(prefix.data(), blockSize, initialCounter, iv);
```

Comment:
Won't this crash/error if the prefix is empty?

Reply:
@mrambacher : I guess you are right, this doesn't look safe to me too.
I think right now it won't crash because for `prefix` there will always be a buffer allocated which is 16 bytes long. So the `prefix` pointer currently seems to point into a 16 bytes buffer, which isn't filled if the input file is empty.
That's potentially also the reason why it worked just fine when I tried it.

Reply:
@mrambacher 

> If the read prefix is empty, perhaps the code should simply return the underlying reader rather than the wrapped one?

The `EncryptedFileSystemImpl`'s methods, e.g. `CreateSequentialCipherStream()` know only about the `provider_`, which is a generic `shared_ptr<EncryptionProvider>`. 
Right now they will call `CreateCipherStream()`, which is a public method on the `EncryptionProvider` interface. The concrete `CTREncryptionProvider` will then internally call its `CreateCipherStreamFromPrefix()` method, which will finally create a `BlockAccessCipherStream` of specialized type `CTRCipherStream`.

If I now wanted to add some special empty file handling in `CreateSequentialCipherStream()`, all I can do is call into the general `EncryptionProvider` interface, which only provides `CreateCipherStream()`. The methods I would acutally need are not on the generic `EncryptionProvider` interface.
So I think it would still be simplest to have the `CTREncryptionProvider` handle empty prefixes itself. This can be made safe by also bypassing the call to `decodeCTRParameters()` in case the prefix is fully empty.

So I have adjusted the PR accordingly, as I don't know how to return some `CTREncryptionProvider` internals when inside the `EncryptedFileSystemImpl`.
The adjusted version still fixes the problem of not reporting corruption for 0 byte files, as I have just verified when calling it from our application.

I also added a unit for reading an empty file with the default env and the encrypted env.
The unit test will fail for the encrypted env without the changes in this PR, and will pass with them.